### PR TITLE
Hid shape fields

### PR DESCRIPTION
**Pull request by anqitong**
#62

### CODE REVIEW

**Comment by anqitong** [comment]
@Thomas-dot-G  : can you test it? Only the console area is not implemented yet. You need to fill the fields for Pencil. I'm working on the console area now.

File: src/controller/guiController/AddToConsoleController.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	private JTextField textValue;
+	
+	
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		ArrayList<Shape> shapes = new ArrayList<Shape>();
+		
+		//setting pencil parameters
+		if(thicknessValue.getText()!=null){
+			pencil.setWidth(Integer.parseInt(thicknessValue.getText().trim()));
```
Comment (Thomas-dot-G) [inline src/controller/guiController/AddToConsoleController.java:169]: can you comment the trim function ? (I don't know what it does)

File: src/controller/guiController/AddToConsoleController.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		}
+		String pencilRed = pencilRedValue.getText();
+		String pencilGreen = pencilGreenValue.getText();
+		String pencilBlue = pencilBlueValue.getText();
+		if(!pencilRed.equals("") && !pencilGreen.equals("") && !pencilBlue.equals("")){
+			int red = Integer.parseInt(pencilRed.trim());
+			int green = Integer.parseInt(pencilGreen.trim());
+			int blue = Integer.parseInt(pencilBlue.trim());
+			int rgb = red;
+			rgb = (rgb << 8) + green;
```
Comment (Thomas-dot-G) [inline src/controller/guiController/AddToConsoleController.java:186]: comment also please

File: src/controller/guiController/ControlRectangleRadioBtn.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+//			this.modele.setX(Integer.parseInt(rectValueX.getText().trim()));
+//		}
+//		if(rectValueY.getText()!=null){
+//			this.modele.setY(Integer.parseInt(rectValueY.getText().trim()));
+//		}
+//		if(rectWidthValue.getText()!=null){
+//			this.modele.setWidth(Integer.parseInt(rectWidthValue.getText().trim()));
+//		}
+//		if(rectHeightValue.getText()!=null){
+//			this.modele.setHeight(Integer.parseInt(rectHeightValue.getText().trim()));
```
Comment (Thomas-dot-G) [inline src/controller/guiController/ControlRectangleRadioBtn.java]: Why is there some commented code here ?

**Comment by Thomas-dot-G** [comment]
Very nice

For how it works, I would have done something else, because now you can't really add several shapes.
When clicking on add, it should just display something in the console (I know that is normal now if it doesn't), but should not open and create the svg image, only the run button should trigger that part

Otherwise, nice !!

**Comment by Thomas-dot-G** [comment]
Could you add some padding also around the window please

File: src/controller/guiController/AddToConsoleController.java (hunk 1 above)
Comment (anqitong) [inline src/controller/guiController/AddToConsoleController.java:169]: yes I will

File: src/controller/guiController/AddToConsoleController.java (hunk 2 above)
Comment (anqitong) [inline src/controller/guiController/AddToConsoleController.java:186]: yes I will comment more

**Comment by anqitong** [comment]
Corrected everything.
You CAN add several shapes.

**Comment by Thomas-dot-G** [comment]
Sometimes, things don't appear on the final display of the image.
Polygon and polyline do not display the points in the xml.
You should disable the fill radio box for lines, polylines and text.
Add a scroll may be on the left, because some times I can't access the add button.
You can also put default value for the pencil of display a warning if the fields are empty.
And the `run` button same an image, but we can't click it again, because the image is already created (we can click but nothing will be done).
Run should just show a preview with a random name, maybe in awt, and a button save should save it in xml with a name that the user has to put.

**Comment by anqitong** [comment]
@Thomas-dot-G : I fixed all of that. Please check it and tell if it is okay

**Comment by Thomas-dot-G** [comment]
An ellipse create a circle (and with the wrong argument it does not show up.
Then when you exit the preview page, it also exit the whole GUI, which is quite problematic.

File: src/controller/guiController/AddToConsoleController.java [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+			
+			/* 
+			 * convert the red, green and blue integers into a rgb integer
+			 */
+			int red = Integer.parseInt(pencilRed.trim());
+			int green = Integer.parseInt(pencilGreen.trim());
+			int blue = Integer.parseInt(pencilBlue.trim());
+			int rgb = red;
+			rgb = (rgb << 8) + green;
+			rgb = (rgb << 8) + blue;
```
Comment (Thomas-dot-G) [inline src/controller/guiController/AddToConsoleController.java:187]: I still am not sure what does this ligne with the `<<`

**Comment by anqitong** [comment]
Does not seem so problematic to me, the user does not have to close the AWT window.

**Comment by anqitong** [comment]
See the last commit

**Comment by Thomas-dot-G** [comment]
Not sure about the last comment, but  It's not a problem.

However, it can be a bit problematic, when I wanted to preview my image, then it was in front of the GUI, so I quit it to click on save, and I lost everything, twice^^

It means that if you are doing several tests, to see how it goes, you have to keep all the windows open. I think it is problematic.

It is another windows, so instead of making it exit on close, do hide on close, it will be much better (and short to modify)

**Comment by anqitong** [comment]
The problem is the Code in AWTFrame : System.exit(0);
This means it destroys the Java VM so of course it closes my JFrame window too.

**Comment by Thomas-dot-G** [comment]
Replace it with windowEvent.getWindow().dispose();
it is better that doing the same but with hide();

**Comment by anqitong** [comment]
Conflicts fixed : it was due to the renaming of the packages

**Comment by Thomas-dot-G** [comment]
Then I am ok

**Comment by Thomas-dot-G** [comment]
still some conflicts to resolve

**Comment by anqitong** [comment]
Yes

**Comment by starsasumi** [comment]
OK to me

**Pull request merged**